apps/src/templates/sectionsRefresh/QuickAssignTable.jsx

```diff
@@ -1,10 +1,27 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 import PropTypes from 'prop-types';
 import moduleStyles from './sections-refresh.module.scss';
 import i18n from '@cdo/locale';
 import {CourseOfferingCurriculumTypes as curriculumTypes} from '@cdo/apps/generated/curriculum/sharedCourseConstants';
 
-export default function QuickAssignTable({marketingAudience, courseOfferings}) {
+/*
+Represents the (collection of) tables in Curriculum Quick Assign.
+They display side by side as if they were columns, but leaving them
+as independent tables makes styling them simpler.
+*/
+export default function QuickAssignTable({
+  marketingAudience,
+  courseOfferings,
+  updateCourse
+}) {
+  const [assignedCourse, setAssignedCourse] = useState('');
```

Comment:
Could this be passed in as a prop instead to avoid the `useEffect`?

Reply:
I'm not sure about this... for the way the radio buttons and updateSection function work, I think we'll need to keep track of this as state. The way the rest of the new section page currently works is that updateSection is called immediately when a button is pressed (grade is selected, etc) and I was hoping to follow that pattern here. Happy to hear your thoughts though!

Reply:
Also now I'm curious- is useEffect nonideal generally?

Reply:
> is useEffect nonideal generally

No it's definitely got its use! I've just never seen it used quite this way so I'm wondering if there's a different way to do this

Reply:
I think I still want to keep track of it as state (not sure why I feel so strongly- maybe just because of the way it interacts and changes with the radio buttons). Would you feel okay with me passing it in as a prop to use as the default when the state is initialized?

Reply:
What I'm suggesting is to have a prop to `QuickAssignTable` and `CurriculumQuickAssign` that is `section.course`. Then the onclick handler could call `updateCourse(assignedCourse)` and update `section` without the `useEffect`. Definitely might be missing something though!

Reply:
(also happy to talk this through in a call if you'd like!)

Reply:
(sorry, our comments crossed)

Reply:
I think I'm just confused as to why the state needs to be duplicated

Reply:
How would the radio buttons know which one is selected though? I am happy to move the updateSection right into the onClick, but then I would still have to keep track of it in state anyway right?

Reply:
(We had a conversation offline) Thank you for the idea! I understand now, and have passed through Section.course, which is now an object with displayName as the useful property in this case. I was able to eliminate the state and useEffect completely, and adjusted the tests and radio/save functionality accordingly!